config/main.py

```diff
@@ -314,6 +308,78 @@ def is_interface_bind_to_vrf(config_db, interface_name):
         return True
     return False
 
+# TODO move to a common file like sonic_device_util.py
+# Validate whether a given namespace name is valid in the device.
+def validate_namespace(namespace):
+    if not sonic_device_util.is_multi_npu():
+        return True
+
+    namespaces = sonic_device_util.get_all_namespaces()
+    if namespace in namespaces['front_ns'] + namespaces['back_ns']:
+        return True
+    else:
+        return False
+
+# Get the table name based on the interface type
+def get_port_table_name(interface_name):
+    """Get table name by port_name prefix
+    """
+    if interface_name.startswith(SONIC_INTERFACE_PREFIXES["Ethernet"]):
+        if VLAN_SUB_INTERFACE_SEPARATOR in interface_name:
+            return "VLAN_SUB_INTERFACE"
+        return "PORT"
+    elif interface_name.startswith(SONIC_INTERFACE_PREFIXES["PortChannel"]):
+        if VLAN_SUB_INTERFACE_SEPARATOR in interface_name:
+            return "VLAN_SUB_INTERFACE"
+        return "PORTCHANNEL"
+    elif interface_name.startswith(SONIC_INTERFACE_PREFIXES["Vlan"]):
+        return "VLAN_INTERFACE"
+    elif interface_name.startswith(SONIC_INTERFACE_PREFIXES["Loopback"]):
+        return "LOOPBACK_INTERFACE"
+    else:
+        return ""
+
+# Return the namespace where an interface belongs
```

Comment:
Add TODO here to move this function to sonic-py-common

Reply:
It seems this function still makes sense in sonic-py-common, no?

Reply:
Sure Joe, there are couple of functions which I was planning move, the two you suggested, and other API's like validate_namespace(). But I was looking to do that movements after this PR got merged, if that is ok with you.

Reply:
This particular function get_port_namespace(), I am thinking of moving to utilitiles-common in sonic-utilities as it uses the alias mode and has dependency on cliCommon module.